tests/test_lesion.py

```diff
@@ -0,0 +1,211 @@
+            RandFlipd,
+            RandRotated,
+            RandAdjustContrastd,
+            ToTensorD,
+        ]
+
+    for t, e in zip(output_transform.transforms, expected_transfom):
+        assert isinstance(t, e)
+
+
+@pytest.mark.parametrize("transform_name", [("2d_ct_transform")])
+def test_get_pretesting_transforms(transform_name, lesion_obj):
+    """
+    Tests the get_pretesting_transforms function.
+    verifies the functionality by comparing the output transforms with the expected transforms based on the provided transform_name.
+
+    Parameters:
+    ----------
+    transform_name (string): "2d_ct_transform".
+    lesion_obj (object): The lesion object that provides the get_pretesting_transforms method.
+    ----------
+    """
+    output_transform = lesion_obj.get_pretesting_transforms(transform_name)
+    assert isinstance(output_transform, Compose)
+
+    if transform_name == "2d_ct_transform":
+        expected_transfom = [
+            LoadImageD,
+            EnsureChannelFirstD,
+            ScaleIntensityRanged,
+            ToTensorD,
+        ]
+
+    for t, e in zip(output_transform.transforms, expected_transfom):
+        assert isinstance(t, e)
+
+
+@pytest.mark.parametrize("transform_name", [("2d_ct_transform")])
+def test_get_postprocessing_transforms(transform_name, lesion_obj):
+    """
+    Tests get_postprocessing_transforms function.
+    verifies the functionality by comparing the output transforms with the expected transforms based on the provided transform_name.
+
+    Parameters:
+    ----------
+    transform_name (string):"2d_ct_transform".
+    lesion_obj (object): The lesion object that provides the get_postprocessing_transforms function.
+    ----------
+    """
+    output_transform = lesion_obj.get_postprocessing_transforms(transform_name)
+    assert isinstance(output_transform, Compose)
+
+    if transform_name == "2d_ct_transform":
+        expected_transfom = [ActivationsD, AsDiscreteD, FillHolesD, RemoveSmallObjectsD]
+
+    for t, e in zip(output_transform.transforms, expected_transfom):
+        assert isinstance(t, e)
+
+
+def test_predict_2dto3d(lesion_obj):
+    """
+    Tests the prediction function for lesion object.
+    verifies the functionality by performing predictions on a 3D volume (with size 64,64,...).
+
+    Parameters:
+    ----------
+    lesion_obj (object): The lesion object that provides the predict_2dto3d function.
+    ----------
+    """
+
+    liver_mask = segment_liver(
+        prediction_path=config.test["test_volume"],
+        modality="CT",
+        inference="sliding_window",
+        cp_path=config.test["reference_sliding_window"],
+    )
+    lesion_inference = "3D"
+    volume_dir = "tests/testdata/resized_liver/volume/resized_liver.nii"
+    close = MorphologicalClosing(iters=4)
+    prediction = lesion_obj.predict_2dto3d(
+        volume_path=volume_dir,
+        liver_mask=close(liver_mask[0]).permute(3, 0, 1, 2)
+        if lesion_inference == "3D"
```

Comment:
you can remove the unnecessary if conditions, since lesion_inference is hardcoded anyways.